src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -1215,6 +1215,9 @@ void EmitBackreferenceConditional(RegexNode node)
                 RegexNode? noBranch = node.ChildCount() > 1 && node.Child(1) is { Type: not RegexNode.Empty } childNo ? childNo : null;
```

Comment:
Just curious, comparing this to EmitExpressionConditional 1374 it seems we cannot assume a 2nd child here, but we can below?
also when can node.ChildCount() be > 2, why do we not test node.ChildCount() == 2?

Reply:
ah "we want all engines to be able to trust that every Testref will have two children," -- does that mean this can be `RegexNode? noBranch = node.Child(1) is { Type: not RegexNode.Empty } childNo ? childNo : null;`

Reply:
also perhaps this method could have an assert that child count == 2, not greater. And EmitExpressionConditional can assert child count ==3.

Reply:
The `node.ChildCount() > 1 && ` isn't necessary.  I'll delete it.  I thought I had. 

As for asserts, we assert that condition in a variety of places,  including as part of validating the whole tree coming out of the parser.  The code would get a lot noisier if we needed to re-assert those conditions on every use,  e.g. that every Concatenate has at least two children. I'd prefer to leave it out here.